### ISSUE TITLE
shallow does not use existing context

### ISSUE THREAD

**Issue by sylvain-hamel** [airbnb/enzyme#1053]
I use `recompose` to set up my context. I know I can pass `options.context` to both `shallow` and `mount` but in my case setting up the context is complex and it's much more convenient to use `withContext`.

The unit under test is a component called `Component` which uses `foo` from the context. 

The unit test creates a context this way.

```
// use recompose's withContext to setup the context
const ComponentWithContext = witContext(
  {foo: PropTypes.string}, ()=>{foo: 'test'}
)(Component);

const wrapper = shallow(<ComponentWithContext/>);
```

When I `mount` my component, this works fine; the context is available to the `Component`. But when I use `shallow` the context is empty.

 Why doesn't it work with `shallow` as well?

**Comment by ljharb** [comment]
I believe this is a duplicate of #664, assuming that `getChildContext` is how `withContext` is implemented.

But also, since `withContext` is an HOC, you need `shallow(<ComponentWithContext />).dive()` as well.

**Comment by ljharb** [comment]
Fixed in #664.
